tests/helper/prism-loader.ts

```diff
@@ -108,18 +110,28 @@ export function createPrismDOM (): PrismDOM<{}> {
 		const g = global as unknown as Record<string, unknown>;
 		let undo;
 		try {
-			undo = overwriteProps(g, {
+			const globals = {
 				window,
 				document: window.document,
-				navigator: window.navigator,
 				location: window.location,
 				getComputedStyle: window.getComputedStyle,
 				setTimeout: noop,
+			};
+
+			// Set up navigator separately to make it writable (it's read-only in Node.js)
+			Object.defineProperty(g, 'navigator', {
+				value: window.navigator,
+				configurable: true,
+				writable: true,
 			});
+
+			undo = overwriteProps(g, globals);
 			fn();
 		}
 		finally {
 			undo?.();
+			// Clean up navigator property
+			delete (g as any).navigator;
```

Comment:
You can do a little better here like: 
```ts
			delete (g as Partial<typeof global>).navigator;
```

But really it's not necessary, casting to `any` is totally fine here.

Reply:
Clever!